src/map.rs

```diff
@@ -194,6 +199,90 @@ impl<'a, K: Ord + 'a, V: 'a> ManagedMap<'a, K, V> {
             &mut ManagedMap::Owned(ref mut map) => map.remove(key)
         }
     }
+
+    /// ManagedMap contains no elements?
+    pub fn is_empty(&self) -> bool {
+        match self {
+            &ManagedMap::Borrowed(ref pairs) =>
+                pairs.iter().all(|item| item.is_none()),
+            #[cfg(any(feature = "std", feature = "alloc"))]
+            &ManagedMap::Owned(ref map) =>
+                map.is_empty()
+        }
+    }
+
+    /// Returns the number of elements in the ManagedMap.
+    pub fn len(&self) -> usize {
+        match self {
+            &ManagedMap::Borrowed(ref pairs) =>
+                pairs.iter()
+                .take_while(|item| item.is_some())
+                .count(),
+            #[cfg(any(feature = "std", feature = "alloc"))]
+            &ManagedMap::Owned(ref map) =>
+                map.len()
+        }
+    }
+
+    /// Short-cut for `into_iter()`
```

Comment:
Why is this a shortcut for `into_iter`? Does the standard library do the same?

Reply:
You're right: [in BTreeMap](https://doc.rust-lang.org/stable/src/alloc/btree/map.rs.html#1136) it is the other way around.